### ISSUE TITLE
Don't commit JARs in .test.dependencies/lib

### ISSUE THREAD

**Issue by chanseokoh** [GoogleCloudPlatform/google-cloud-eclipse#2200]
With #2190 merged, we no longer have to commit the JARs.

mockito-core-1.10.10.jar
objenesis-2.2.jar
hamcrest-core-1.1.jar

**Comment by chanseokoh** [comment]
![selection_026](https://user-images.githubusercontent.com/10523105/28678577-bb3bdc6c-72be-11e7-9ee8-6cdc299277d7.png)

I noticed that the bundle does not have hamcrest-core-1.1.jar in its classpath. Should we add it? It's strange we don't have any build error even if they are not in the classpath. (On the other hand, removing mockito-core from the classpath results in thousands of compile errors.) @briandealwis are we correctly picking up hamcrest-core in our code?

**Comment by briandealwis** [comment]
Weird: `lib/` should contain only `.gitignore` and `mockito-core-1.10.19.jar` and `objensis-2.2.jar` should have been removed from the git repo.

Maybe `.test.dependencies` is better to use `dependency:copy` with an explicit list of `<artifactItems>` rather than pulling in transitive dependencies.  That will avoid the confusion of having ignored jars.  I'll put together a patch.

We don't want to add `hamcrest-core` 1.1.
Hamcrest 1.1 is an interesting case.  `mvn dependency:tree` shows that `hamcrest-core` 1.1 is brought it via `mockito-core`.   But Mockito is an OSGi bundle and if you look at its manifest you'll see an `Import-Package: org.hamcrest;version="[1.0,3.0)"`.  Since it's happy with later versions of Hamcrest, we just use Hamcrest 1.3.

Note that Mockito's `Import-Package` is actually one of the reasons we created this `.test.dependencies` bundle in the first place as there is only one bundle that offers a package `org.hamcrest` _without_ mandatory attributes: the broken `org.hamcrest 1.1` from Orbit.  All other bundles require mandatory attributes to pull in the hamcrest packages because they're split packages.

Why is Orbit's `org.hamcrest` 1.1.0 bundle broken?  It's a facade bundle that uses `Require-Bundle` to bring in various org.hamcrest.* bundles but it uses loose bounds and so can (and does) bring in Hamcrest 1.3 bundles:
```
Require-Bundle: org.hamcrest.core;bundle-version="1.1.0";visibility:=reexport,
  org.hamcrest.library;bundle-version="1.1.0";visibility:=reexport,
  org.hamcrest.generator;bundle-version="1.1.0";resolution:=optional;visibility:=reexport,
  org.hamcrest.integration;bundle-version="1.1.0";visibility:=reexport,
  org.hamcrest.text;bundle-version="1.1.0";visibility:=reexport
```
But the bundle then re-exports the classes in `org.hamcrest` as 1.1, even though they may actually be from a different version:
```
Export-Package: org.hamcrest;version="1.1.0"
```
